### ISSUE TITLE
Turning layers with affine into layers with BN

### ISSUE THREAD

**Issue by Galisela** [davisking/dlib#1091]
Hi,

I am new to c++, but the clear documentation actually allows me to use dlib with little understanding of the syntax, so thanks for that!

I manage to easily load trained networks with BN into testing networks with affine transform, however I am interested in doing transfer learning for resnet and was wondering if there was a similar way to assign layers with constant affine back into batch normalization.

Thanks!

**Comment by davisking** [comment]
There isn't a way to do this. Instead, you should be careful to always save the network with BN layers. Once it's converted to affine there is no converting it back.  At least not with the current code.  This is because the conversion to affine drops some of the BN state and to convert back you would have to somehow guess at that state or recompute it from the training data, which is not impossible, but included in the dlib code since the process is much more confusing than just saving the network with BN layers in the first place.

**Comment by Galisela** [comment]
Thank you! I assumed this much.
One other tiny question. Do you happen to have your trained resnet34 model saved with bn layers?
That would be amazing.

**Comment by davisking** [comment]
No, I unfortunately overwrote the file with the affine only version :/

**Comment by Galisela** [comment]
Ok, was worth a try. Thanks.

**Comment by davisking** [comment]
I doubt fine tuning it's a good idea anyway.  You should use something like a k-nearest-neighbor or linear SVM on top of it instead.
